### ISSUE TITLE
Local storage in sortabletable.js

### ISSUE THREAD

**Issue by c16jonha** [HGustavs/LenaSYS#4377]
Local storage in sortabletable.js is currently storing the column filter

~~~
// Private array that contains names of filtered columns
				columnfilter = JSON.parse(localStorage.getItem(tableid+"_filtercolnames"));
~~~

If another table has been previously used, it remembers columns that does not appear in the current table.

Some kind of check for non-existing columns in the filter is necessary.

if one column in the filter is unknown then set column filter to table head column list.

**Comment by d15marwi** [comment]
This issue is solved

**Comment by b16nilpe** [comment]
Column filter is stored in separate entries based on tableID which means that filtering in one access table will be applied to all access tables but a change in access won't affect a table in results. 
As of now the only table with dynamic columns is the result table. The header IDs in the result table is based on Lids which should be somewhat unique which, hence the filter status stored in local storage will also be somewhat unique for columns in different courses. This was implemented/fixed in #5104.

The following problem remains an issue:
If there is a record in local storage, new columns are not shown by default. This presents a real problem in the result table with the dynamic columns. If the result table of one course is visited, all other courses will fail to display their columns since there is a record in local storage.

**Comment by c16matan** [comment]
Merged. Closing.
